test/avocado/seleniumlib.py

```diff
@@ -135,16 +139,21 @@ def wait(self, method, text, baseelement, overridetry, fatal, cond):
     overridetry - change value of repeats
     fatal - boolean if search is fatal or notice
     cond - use selenim conditions (aliases are defined above class)
+    jscheck - use javascipt to wait for element has attribute-data loaded, it is safer, but slower
         """
         if not baseelement:
             baseelement = self.driver
         returned = None
         cond = cond if cond else visible
         internaltry = overridetry if overridetry else self.default_try
+        self.__internal_function_used = lambda :WebDriverWait(baseelement, self.default_explicit_wait).until(cond((method, text)))
```

Comment:
self.__internal_function_used feels a bit hackish, since it's not obvious what happens.
Can we pass it where needed or use a contextmanager to make sure it's only set when needed? (e.g. https://github.com/cockpit-project/cockpit/blob/master/test/testvm.py#L52)